src/app/pages/tasks/legend/component.ts

```diff
@@ -21,11 +37,31 @@ export class TaskLegendComponent {
       value: '-',
     },
     {
-      descriptionShort: 'No longer available',
+      descriptionShort: 'Unavailable',
       completedOn: false,
       active: false,
       earnedPoints: '-',
       value: '-',
     },
   ];
+
+  public setSelector(i: number) {
+    this.selected = i;
+    this.selectedTaskType.emit(this.tasks[i]);
+  }
+
+  public setScores(scores: any[]) {
+    scores.map(
+      (function (obj: any) {
+        function setScore(taskScore: any) {
+          const index = obj.tasks.findIndex(
+            (task) => task.descriptionShort == taskScore.descriptionShort
+          );
+          obj.tasks[index].value = taskScore.value;
+          obj.tasks[index].earnedPoints = taskScore.earnedPoints;
+        }
+        return setScore;
+      })(this)
+    );
```

Comment:
This syntax is awkward. In all the places you return a function like this, it is more conventional to use the fat arrow syntax. For example:

```suggestion
    scores.map((task) => {
      const index = this.tasks.findIndex(
        (t) => t.descriptionShort === task.descriptionShort
      );
      this.tasks[index].value = task.value;
      this.tasks[index].earnedPoints = task.earnedPoints
    });
```